arrow-array/src/array/byte_view_array.rs

```diff
@@ -638,7 +647,9 @@ where
             assert_eq!(views_builder.len(), len);
             views_builder.finish()
         } else {
-            // TODO: the first u32::MAX can still be reused
+            // otherwise, create a new buffer for large strings
+            // TODO: the original buffer could still be used
+            // until the offset reaches `u32::max`.
```

Comment:
```suggestion
            // Otherwise, create a new buffer for large strings
            // TODO: the original buffer could still be used
            // by making multiple slices of u32::MAX length
```

You actually don't need to copy the data ever

Reply:
That is a clever idea 💯 

FYI @XiangpengHao